docs/reference/modules/cluster/shards_allocation.asciidoc

```diff
@@ -57,6 +57,7 @@ one of the active allocation ids in the cluster state.
       the same shard on a single host, based on host name and host address.
       Defaults to `false`, meaning that no check is performed by default. This
       setting only applies if multiple nodes are started on the same machine.
+      Will perform by this setting apply when "Same host name" or "Same address".
```

Comment:
```
    If `true`, prevents allocation of multiple copies of the same shard to nodes
    that have the same host name or the same host address since this indicates
    that those nodes are running on the same host machine. Defaults to `false`,
    meaning that multiple copies of the same shard may possibly be allocated to
    multiple nodes running on the same host machine. This setting only has an
    effect if you are running multiple nodes on the same host machine.
```

Reply:
Actually, due to the `else if` [here](https://github.com/elastic/elasticsearch/blob/26c86871fc091900952e88e252c36fbfedf8d5fa/server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/SameShardAllocationDecider.java#L100), the hostName will be processed only when hostAddress is empty. I'm not sure if this is what the contributor wants, but IMO it's a bug.